Libplanet/Store/Trie/Trie.cs

```diff
@@ -0,0 +1,328 @@
+
+                case HashNode hashNode:
+                    var hn = GetNode(hashNode.HashDigest);
+                    return Insert(hn, prefix, key, value);
+
+                default:
+                    throw new InvalidTrieNodeException("Not supported node came." +
+                                                       $" raw: {node.ToBencodex().Inspection}");
+            }
+        }
+
+        private bool TryGet(
+            INode node,
+            ImmutableArray<byte> prefix,
+            ImmutableArray<byte> path,
+            out IValue value)
+        {
+            switch (node)
+            {
+                case null:
+                    value = null;
+                    return false;
+
+                case ValueNode valueNode:
+                    value = valueNode.Value;
+                    return true;
+
+                case ShortNode shortNode:
+                    if (path.Length < shortNode.Key.Length
+                        || !path.Take(shortNode.Key.Length).SequenceEqual(shortNode.Key))
+                    {
+                        value = null;
+                        return false;
+                    }
+
+                    return TryGet(
+                        shortNode.Value,
+                        prefix.AddRange(path.Take(shortNode.Key.Length)),
+                        path.Skip(shortNode.Key.Length).ToImmutableArray(),
+                        out value);
+
+                case FullNode fullNode:
+                    INode childNode = fullNode.Children[path[0]];
+                    return TryGet(
+                        childNode,
+                        prefix.Add(path[0]).ToImmutableArray(),
+                        path.Skip(1).ToImmutableArray(),
+                        out value);
+
+                case HashNode hashNode:
+                    try
+                    {
+                        INode resolvedNode = GetNode(hashNode.HashDigest);
+                        return TryGet(resolvedNode, prefix, path, out value);
+                    }
+                    catch (KeyNotFoundException)
+                    {
+                        value = null;
+                        return false;
+                    }
+
+                default:
+                    throw new InvalidTrieNodeException(
+                        $"Invalid node: raw: {node.ToBencodex().Inspection}");
+            }
+        }
+
+        /// <summary>
+        /// Gets the node corresponding to <paramref name="nodeHash"/> from storage,
+        /// (i.e., <see cref="KeyValueStore"/>).
+        /// </summary>
+        /// <param name="nodeHash">The hash of node to get.</param>
+        /// <returns>The node corresponding to <paramref name="nodeHash"/>.</returns>
+        private INode GetNode(HashDigest<SHA256> nodeHash)
+        {
+            return NodeDecoder.Decode(Codec.Decode(KeyValueStore.Get(nodeHash.ToByteArray())));
+        }
+
+        // TODO: Support secure trie.
```

Comment:
Is it still valid TODO?

Reply:
No, it was resolved by aeff00f. I will remove the comment. Thanks.